src/scripts/main.js

```diff
@@ -1,3 +1,317 @@
+    const rows = [...document.querySelectorAll('table tbody tr')];
+
+    rows.sort((a, b) => {
+      const aCell = a.children[indexOfColumn].textContent.trim();
+      const bCell = b.children[indexOfColumn].textContent.trim();
+
+      const aValue = parseFloat(aCell.replace(/[$,]/g, '')) || aCell;
+      const bValue = parseFloat(bCell.replace(/[$,]/g, '')) || bCell;
+
+      let result;
+
+      if (!isNaN(aValue) && !isNaN(bValue)) {
+        result = aValue - bValue;
+      } else {
+        result = aValue.localeCompare(bValue);
+      }
+
+      return isAscending ? result : -result;
+    });
+
+    rows.forEach((row) => {
+      tbody.appendChild(row);
+    });
+  };
+}
+
+function getFormData(form) {
+  const data = new FormData(form);
+  const nameInput = data.get('name');
+  const position = data.get('position');
+  const office = data.get('office');
+  const age = data.get('age');
+  const salary = +data.get('salary');
+
+  return {
+    name: nameInput,
+    position,
+    office,
+    age,
+    salary: `$${salary.toLocaleString('en-US')}`,
+  };
+}
+
+function addUser(e) {
+  e.preventDefault();
+
+  const form = document.querySelector('.new-employee-form');
+  const tbody = document.querySelector('tbody');
+
+  const user = getFormData(form);
+
+  if (user.name.length < 4 || !user.name.trim().length) {
+    createNotification(
+      'error',
+      'Fix your name',
+      'The name must consist of at least 4 letters.',
+    );
+
+    return;
+  }
+
+  if (!user.position.trim().length) {
+    createNotification(
+      'error',
+      'Fix your position',
+      'Position field can not be empty.',
+    );
+
+    return;
+  }
+
+  if (user.office === null) {
+    createNotification(
+      'error',
+      'Fix your office',
+      'Ofice field can not be empty.',
+    );
+
+    return;
+  }
+
+  if (user.age < 18 || user.age > 90) {
+    createNotification('error', 'Fix your age', 'Your age is not appropriate.');
+
+    return;
+  }
+
+  if (!parseFloat(user.salary.replace(/[$,]/g, ''))) {
+    createNotification(
+      'error',
+      'Fix your salary',
+      'Salary field can not be empty',
+    );
+
+    return;
+  }
+
+  const newTr = document.createElement('tr');
+
+  Object.values(user).forEach((value) => {
+    const td = document.createElement('td');
+
+    td.innerHTML = value;
+
+    newTr.appendChild(td);
+  });
+
+  tbody.appendChild(newTr);
+  form.reset();
+
+  const select = form.querySelector('select[name="office"]');
+
+  select.selectedIndex = 0;
+  createNotification('success', 'Success!', 'User was successfully added.');
+}
+
+function createNotification(state, fix, message) {
+  const div = document.createElement('div');
+  const description = document.createElement('p');
+  const title = document.createElement('p');
+
+  div.dataset.qa = 'notification';
+  div.classList.add(state, 'notification');
+  title.innerHTML = fix;
+  title.className = 'title';
+  description.innerHTML = message;
+
```

Comment:
The `createNotification` function appends the description before the title, which might affect the display order. Consider appending the title first, followed by the description.